src/main/java/io/apicurio/datamodels/core/validation/rules/other/SecurityRequirementUniquenessValidationRule.java

```diff
@@ -0,0 +1,134 @@
+       
+        // first check the names of the security requirements we need to compare
+        List<String> currentSecurityRequirementNames = currentSecurityRequirement
+                .getSecurityRequirementNames();
+        List<String> otherSecurityRequirementNames = otherSecurityRequirement.getSecurityRequirementNames();
+        // sort the lists before checking if they are equal
+        Collections.sort(currentSecurityRequirementNames);
+        Collections.sort(otherSecurityRequirementNames);
+        
+        // check if they are equal
+        if (currentSecurityRequirementNames.equals(otherSecurityRequirementNames)) {
+            int matchesFound = 0;
+            // if they have a list of the same names, we need to check the scopes for each security
+            // requirement name against
+            // the scopes for all of the other security requirements - who have the same name (to keep
+            // context).
+            for (int i = 0; i < currentSecurityRequirementNames.size(); i++) {
+                String secReqName = currentSecurityRequirement.getSecurityRequirementNames().get(i);
+                
+                for (int j = 0; j < otherSecurityRequirementNames.size(); j++) {
+                    String otherSecReqName = otherSecurityRequirement.getSecurityRequirementNames().get(j);
+                    if (secReqName.equals(otherSecReqName)) { // if they have the same name eg: [pet_auth] equals [pet_auth]
+                        // now get the scopes for each one of these sr names eg: ["read:pets", "write:pets"] ["write:pets","read:pets"]
+                        List<String> secReqScopes = currentSecurityRequirement.getScopes(secReqName);
+                        List<String> otherSecReqScopes = otherSecurityRequirement.getScopes(otherSecReqName);
+                        // sort them
+                        Collections.sort(secReqScopes);
+                        Collections.sort(otherSecReqScopes);
+                        if (secReqScopes.equals(otherSecReqScopes)) { // compare them
+                            matchesFound += 1; // increment this variable when the scopes match
+                        }
+                    }
+                }
+            }
+            // if the number of matches found == the number of sec requirements for the current one
+            if (matchesFound == currentSecurityRequirementNames.size()) {
+                // Then we have a duplicated security requirement
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @param currentSecurityRequirement
+     * @param allRequirements
+     * @return true if duplicates are found
+     */
+    private boolean checkForDuplicates(SecurityRequirement currentSecurityRequirement,
+            List<SecurityRequirement> allRequirements) {
+        ArrayList<SecurityRequirement> clonedList = new ArrayList<SecurityRequirement>();
```

Comment:
I think you can pass in `allRequirements` to the `ArrayList` constructor and then you don't need to call `addAll` below.  Check on that though - don't take my word for it. :)

Reply:
👍